central/group/datastore/internal/store/store_impl.go

```diff
@@ -139,12 +139,12 @@ func addInTransaction(tx *bolt.Tx, group *storage.Group) error {
 	id := group.GetProps().GetId()
 
 	// Check whether the to-be-added group is a default group, ensure that it does not yet exist.
-	defaultGroupExists, err := checkDefaultGroupForProps(tx, group.GetProps())
+	defaultGrp, err := checkDefaultGroupForProps(tx, group.GetProps())
 	if err != nil {
 		return err
 	}
 
-	if defaultGroupExists {
+	if defaultGrp != nil && defaultGrp.GetProps().GetId() != id {
```

Comment:
Why the second part of the boolean expression? If _any_ default group for this auth provider exists, including with the same id, we should not add, no?

Reply:
Yeah, we shouldn't require it in case we are adding groups.